### PR TITLE
Add docker dev environment

### PR DESCRIPTION
**Pull request by uestctxq**

### CODE REVIEW

File: Docker_build.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+## Doris Develop Environment based on docker
```
Comment (imay) [inline Docker_build.md]: you should add ASF license header

**Review by imay — DISMISSED** [review]
you should change your commit log's subject to imperative mood

File: Docker_build.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+4、tar `palo-thirdparty-{date}.tar.gz`，cp `palo-thirdparty-{date}/*` to `/var/local/docker_build/thirdparty/src`, please create `src` if `src` not exist
+
+5、Build docker image
+```aidl
+cd /var/local/docker_build 
+
+docker build -t palo:v1.0  .
+
+-- palo is docker image repository name and base is tag name , you can change them to what you like
```
Comment (kangpinghuang) [inline Docker_build.md]: you should better change palo to doris

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,54 @@
+FROM centos:centos7.5.1804
+
+MAINTAINER tangxiaoqing214445
+
+ENV DEFAULT_DIR /var/local
+
+# add code repository
+ADD thirdparty /var/local/thirdparty
+
```
Comment (EmmyMiao87) [inline Dockerfile]: You should add palo source code in docker image too, otherwise you won't build palo.

File: Docker_build.md (hunk 1 above)
Comment (uestctxq) [reply Docker_build.md]: OK

File: Docker_build.md (hunk 2 above)
Comment (uestctxq) [reply Docker_build.md]: OK

File: Dockerfile (hunk 3 above)
Comment (uestctxq) [reply Dockerfile]: OK

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  
+ENV MAVEN_HOME /usr/share/maven
+  
+# build environment
+WORKDIR ${DEFAULT_DIR}
+
+# build third party 
+RUN /bin/bash thirdparty/build-thirdparty.sh \
+    && ln -s ${DEFAULT_DIR}/thirdparty/installed/bin/thrift /usr/bin/thrift  \
+    && ln -s ${DEFAULT_DIR}/thirdparty/installed/ant/bin/ant /usr/bin/ant  \
```
Comment (morningman) [inline Dockerfile]: We are not using ant any more, using maven instead.

**Review by morningman — CHANGES_REQUESTED** [review]

**Review by morningman — APPROVED** [review]
It works

**Pull request merged**